### PR TITLE
Add NoAuth mode to allow TLS without client authentication

### PR DESCRIPTION
**Pull request by mostynb**

### CODE REVIEW

**Comment by mostynb** [comment]
@rubensf: could you please take a look, or recommend a reviewer?

File: go/pkg/client/client.go [hunk 1]
```diff
@@ -431,6 +431,10 @@ type DialParams struct {
 	// grpc.WithInsecure() is passed in. Should only be used in test code.
 	NoSecurity bool
 
+	// NoAuth is true if TLS is enabled but the client does not need to authenticate
+	// with the server.
```
Comment (rubensf) [inline go/pkg/client/client.go:435]: Maybe comment that NoSecurity takes precendence?

File: go/pkg/flags/flags.go [hunk 2]
```diff
@@ -35,8 +35,10 @@ var (
 	UseRPCCredentials = flag.Bool("use_rpc_credentials", true, "If false, no per-RPC credentials will be used (disables --credential_file, --use_application_default_credentials, and --use_gce_credentials.")
 	// Service represents the host (and, if applicable, port) of the remote execution service.
 	Service = flag.String("service", "", "The remote execution service to dial when calling via gRPC, including port, such as 'localhost:8790' or 'remotebuildexecution.googleapis.com:443'")
-	// ServiceNoSecurity can be set to connect to the gRPC service without TLS.
-	ServiceNoSecurity = flag.Bool("service_no_security", false, "If true, do not use TLS when connecting to the gRPC service.")
+	// ServiceNoSecurity can be set to connect to the gRPC service without TLS and without authentication.
+	ServiceNoSecurity = flag.Bool("service_no_security", false, "If true, do not use TLS or authentication when connecting to the gRPC service.")
+	// ServiceNoAuth can be set to disable authentication while still using TLS.
+	ServiceNoAuth = flag.Bool("service_no_auth", false, "If true, do not authenticate with the service.")
```
Comment (rubensf) [inline go/pkg/flags/flags.go:41]: Similarly, mention NoSecurity takes precendence.

**Review by rubensf — APPROVED** [review]

File: go/pkg/client/client.go (hunk 1 above)
Comment (mostynb) [reply go/pkg/client/client.go]: Pushed a new commit with more explicit wording. It's difficult to make this clear without sounding overly repetetive.

**Review by rubensf — APPROVED** [review]

**Comment by mostynb** [comment]
Thanks- do you need any other changes?

(I'm looking forward to this becoming available in reclient.)

**Comment by rubensf** [comment]
I believe this is good as is :)

But forewarning it'll likely take some time before we update the SDK version on reclient.

**Comment by rubensf** [comment]
Oh, I forgot you likely don't have merge button access.

**Comment by mostynb** [comment]
> But forewarning it'll likely take some time before we update the SDK version on reclient.

No problem.

Thanks!

**Pull request merged**